Source/UI/Controllers/ThreedsWebViewController.swift

```diff
@@ -53,42 +52,67 @@ public class ThreedsWebViewController: UIViewController,
     public override func viewDidLoad() {
         super.viewDidLoad()
 
-        guard let authUrl = url else { return }
-        let myURL = URL(string: authUrl)
-        let myRequest = URLRequest(url: myURL!)
+        guard let authUrl = url,
+              let url = URL(string: authUrl) else {
+            return
+        }
+
+        let myRequest = URLRequest(url: url)
         webView.navigationDelegate = self
         webView.load(myRequest)
     }
+}
 
-    // MARK: - WKNavigationDelegate
+// MARK: - WKNavigationDelegate
+extension ThreedsWebViewController: WKNavigationDelegate {
 
     /// Called when the web view begins to receive web content.
     public func webView(_ webView: WKWebView, didCommit navigation: WKNavigation!) {
-        shouldDismiss(absoluteUrl: webView.url!)
+
+        guard let absoluteUrl = webView.url else {
+            return
+        }
+
+        shouldDismiss(absoluteUrl: absoluteUrl)
     }
 
     /// Called when a web view receives a server redirect.
     public func webView(_ webView: WKWebView,
                         didReceiveServerRedirectForProvisionalNavigation navigation: WKNavigation!) {
         // stop the redirection
-        shouldDismiss(absoluteUrl: webView.url!)
+        guard let absoluteUrl = webView.url else {
+            return
+        }
+
+        shouldDismiss(absoluteUrl: absoluteUrl)
```

Comment:
Do you know if this method and `func webView(_ webView: WKWebView, didCommit navigation: WKNavigation!)` can both be called for a single navigation action? E.g. if the webview tried to load a URL that falls under both delegate methods

As `shouldDismiss` would then be called twice. It's unlikely iOS would call both WK delegate methods but it just popped in my mind.

Reply:
I think they are independent of each other, but the better delegate method here would be `func webView(_ webView: WKWebView, decidePolicyFor navigationAction: WKNavigationAction, decisionHandler: @escaping (WKNavigationActionPolicy) -> Void)`. Do we want to make this change or leave it as is (as far as I can tell the behaviour would be identical)

Reply:
I guess it's not worth it as the behaviour is the same 👍